src/nccl_ofi_freelist.c

```diff
@@ -162,20 +210,32 @@ int nccl_ofi_freelist_add(nccl_ofi_freelist_t *freelist,
 	   structure at the end of the allocation so that large
 	   buffers are more likely to be page aligned (or aligned to
 	   their size, as the case may be). */
-	buffer = malloc(sizeof(struct nccl_ofi_freelist_block_t) +
-			(freelist->entry_size * allocation_count));
-	if (!buffer) {
-		NCCL_OFI_WARN("freelist extension malloc failed: %s", strerror(errno));
-		return -ENOMEM;
+	block_mem_size = freelist_block_mem_size_full_pages(freelist->entry_size, allocation_count);
+	buffer = mmap(NULL, block_mem_size, PROT_READ | PROT_WRITE,
+		    MAP_PRIVATE | MAP_ANON, -1, 0);
+	if (OFI_UNLIKELY(buffer == MAP_FAILED)) {
+		NCCL_OFI_WARN("Unable to map MR buffer (%d %s)",
+			      errno, strerror(errno));
+		buffer = NULL;
```

Comment:
Nit: No need to set `buffer = NULL;` as we are returning function function in next line.